packages/module-utils/src/generate-dts.ts

```diff
@@ -1,8 +1,71 @@
-import type { StateParsedValue, StylableMeta, StylableResults } from '@stylable/core';
+import type {
+    ClassSymbol,
+    ElementSymbol,
+    ImportSymbol,
+    MappedStates,
+    StateParsedValue,
+    StylableMeta,
+    StylableResults,
+} from '@stylable/core';
 
 const SPACING = ' '.repeat(4);
 const asString = (v: string) => JSON.stringify(v);
 
+function addStatesEntries(
+    stateEntries: Map<string, StateParsedValue | string | null>,
+    stStates: MappedStates | undefined
+) {
+    if (stStates) {
+        for (const [stateName, stateDef] of Object.entries(stStates)) {
+            if (typeof stateDef === 'string') {
+                continue;
+            }
+
+            if (!stateEntries.has(stateName)) {
+                stateEntries.set(stateName, stateDef);
+            }
+        }
+    }
+}
+
+function collectLocalStates(cls: ClassSymbol) {
+    const stateEntries = new Map<string, StateParsedValue | null>();
+    let currentClass: ClassSymbol | undefined = cls;
+
+    while (currentClass) {
+        const stStates = currentClass['-st-states'];
+
+        if (stStates) {
+            addStatesEntries(stateEntries, stStates);
+        }
+
+        const extendedClass = currentClass['-st-extends'] as
```

Comment:
why do you need the `as ...`

Reply:
'extendedClass' implicitly has type 'any' because it does not have a type annotation and is referenced directly or indirectly in its own initializer.